src/music/fluidsynth.cpp

```diff
@@ -0,0 +1,157 @@
+/* $Id$ */
+
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file fluidsynth.cpp Playing music via the fluidsynth library. */
+
+#include "../stdafx.h"
+#include "../openttd.h"
+#include "../sound_type.h"
+#include "../debug.h"
+#include "fluidsynth.h"
+#include "midifile.hpp"
+#include <fluidsynth.h>
+
+static struct {
+	fluid_settings_t* settings;    ///< FluidSynth settings handle
+	fluid_synth_t* synth;          ///< FluidSynth synthesizer handle
+	fluid_audio_driver_t* adriver; ///< FluidSynth audio driver handle
+	fluid_player_t* player;        ///< FluidSynth MIDI player handle
+} _midi; ///< Metadata about the midi we're playing.
+
+/** Factory for the FluidSynth driver. */
+static FMusicDriver_FluidSynth iFMusicDriver_FluidSynth;
+
+/** List of sound fonts to try by default. */
+static const char *default_sf[] = {
+	/* Debian/Ubuntu/OpenSUSE preferred */
+	"/usr/share/sounds/sf2/FluidR3_GM.sf2",
+
+	/* RedHat/Fedora/Arch preferred */
+	"/usr/share/soundfonts/FluidR3_GM.sf2",
+
+	/* Debian/Ubuntu/OpenSUSE alternatives */
+	"/usr/share/sounds/sf2/TimGM6mb.sf2",
+	"/usr/share/sounds/sf2/FluidR3_GS.sf2",
+
+	NULL
+};
+
+const char *MusicDriver_FluidSynth::Start(const char * const *param)
+{
+	const char *driver_name = GetDriverParam(param, "driver");
+	const char *sfont_name = GetDriverParam(param, "soundfont");
+	int sfont_id;
+
+	if (!driver_name) driver_name = "alsa";
```

Comment:
Should this perhaps be a configure option? Multiple reasons, first to let package maintainers pick an appropriate default for the system (many will want PulseAudio, I imagine), and second to better support other OS where ALSA does not exist, such as FreeBSD, Windows, and maybe macOS.

Reply:
Possible, but given the hardcoded soundfont locations just above, it's already not going to be "easy" to use on systems other than Linux. Maybe TMWFTLB

Reply:
Right, leave it as-is, and if some package maintainer asks we can make a patch for this. The other alternative is taking the rendered music into buffers and mix internally in our own mixer, and share output that way. That's too much to add to this patch, but something maybe worth pursuing later.